pkg/plugin/sdk/deployment.go

```diff
@@ -206,7 +206,33 @@ func (s *DeploymentPluginServiceServer[Config, DeployTargetConfig]) ExecuteStage
 		stageID:       request.GetInput().GetStage().GetId(),
 		logPersister:  lp,
 	}
-	return executeStage(ctx, s.base, &s.config, nil, client, request, s.logger) // TODO: pass the deployTargets
+
+	dtNames, err := request.GetInput().GetDeployment().GetDeployTargets(s.commonFields.config.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get deploy targets for plugin %s: %v", s.commonFields.config.Name, err)
+	}
+
+	deployTargets := make([]*DeployTarget[DeployTargetConfig], 0, len(dtNames))
+	for _, name := range dtNames {
+		if dt := s.commonFields.config.FindDeployTarget(name); dt != nil {
+			var sdkDt DeployTargetConfig
+			if err := json.Unmarshal(dt.Config, &sdkDt); err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to unmarshal deploy target config: %v", err)
+			}
+
+			deployTargets = append(deployTargets, &DeployTarget[DeployTargetConfig]{
+				Name:   name,
+				Labels: dt.Labels,
+				Config: sdkDt,
+			})
+		}
```

Comment:
[IMO]
We can use the early-return pattern with the error containing the deployTarget name that is not found when `dt == nil` .

Reply:
Fixed on [8173d91](https://github.com/pipe-cd/pipecd/pull/5616/commits/8173d91784e699dcdfdde363e824df4e828f3a5b)